### ISSUE TITLE
Config server - APPROLE authn not working 

### ISSUE THREAD

**Issue by annagalingam** [spring-cloud/spring-cloud-config#2235]
**Describe the bug**
I have configured the Config Server to use Vault as backed and tried to use the authentication mechanism of APPROLE; It neither does any authentication nor connects the Valut. 

I'm using spring boot 3.0.3, Java 17. 

**Sample**
server:
  port: 8888
application.yml
spring:
  cloud:
    config:
      server:
        vault:
          host: myvault.com
          port: 443
          scheme: https
          namespace: test-name
          authentication: APPROLE
          app-role:
            role-id: my-role-id
            secret-id: my-secret-id
            role: app-role
            app-role-path: approle
          kv-version: 2
          backend: kv
  profiles:
    active:
    - vault
  application:
    name: spring-config-server
-- 
Am I missing any other setup?

**Comment by ryanjbaxter** [comment]
I see Spring Vault supports APPROLE but I don't see it documented in Spring Cloud Vault which is what Spring Cloud Config uses.  @mp911de is it supported in Spring Cloud Vault?

**Comment by annagalingam** [comment]
Now getting below error when get secret-id API is called. I analyzed and found that the "X-Vault-Namespace" header is not being passed in the request hence this error from the vault. 

403 Forbidden: "{"errors":["1 error occurred:\n\t* permission denied\n\n"]}<EOL>"

But I see https://github.com/spring-cloud/spring-cloud-config/pull/1566 PR has fixed that namespace issue. But still, I'm getting the error. 

I added "spring-vault-core" dependency in my pom.

 Am I missing something here?

**Comment by mp911de** [comment]
Yes, Spring Cloud Vault supports AppRole authentication. Configuring this authentication mechanism comes with a bit of complexity as there are push/pull modes associated with SecretId and RoleId.

The code is at https://github.com/spring-cloud/spring-cloud-vault/blob/c8f2168691465271652619d2c9d8d70c80554ab2/spring-cloud-vault-config/src/main/java/org/springframework/cloud/vault/config/ClientAuthenticationFactory.java#L206-L258.

You can find the reference docs for Spring Cloud Vault AppRole auth at https://docs.spring.io/spring-cloud-vault/docs/current/reference/html/#approle-authentication

**Comment by annagalingam** [comment]
@mp911de  Yes followed the same document. 

I'll share all my config again. Pls check and let me know if I am missing anything here. 

pom.xml 

![image](https://user-images.githubusercontent.com/82989087/222458816-310c5ec2-8f3a-4b3e-9d86-8469b0fb8400.png)


application.xml
![image](https://user-images.githubusercontent.com/82989087/222458289-2ac1ebc7-8281-41b1-a1e1-4e32afb05bcf.png)


Java
![image](https://user-images.githubusercontent.com/82989087/222457734-e8155ff0-10e5-4ed1-b6df-eabf6969705e.png)

Error: 
org.springframework.vault.authentication.VaultLoginException: Cannot get Role id using AppRole: 1 error occurred:
	* permission denied

I analyzed and found that the "X-Vault-Namespace" header is not being passed in the request hence this error from the vault.

Do you think I should include something here?

Apologies for the image.
